src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp

```diff
@@ -37,7 +37,10 @@ void TargetedMovementGeneratorMedium<T, D>::_setTargetLocation(T* owner, bool up
         return;
 
     if (owner->GetTypeId() == TYPEID_UNIT && !i_target->isInAccessiblePlaceFor(owner->ToCreature()))
+    {
+        owner->ToCreature()->SetCannotReachTarget(true);
```

Comment:
TargetedMovementGenerator was already aware of whether a path could be found, it just didn't do much with it.